### ISSUE TITLE
Unable to start Blockchain - after reinstall defi-wallet

### ISSUE THREAD

**Issue by baludwi** [DeFiCh/app#1007]
#### Actual behavior:
After start deli-wallet the error "unable to start blockchain" appears

#### Expected behavior:
App should start normally

#### What are your environment parameters?

- DeFi Wallet Version: 2.5.2
- Operating System (OS): Mac OS 
- Current free storage capacity: 

#### Anything else we need to know?
I have deinstalled the wallet and dropped all installation files. After reinstall defi-wallet again, i got the error message "unable to start blockchain". In App.log I can find the message which seems to be the reason for this behaviour. 

[App.log](https://github.com/DeFiCh/app/files/6892974/App.log)
![Error](https://user-images.githubusercontent.com/47607779/127321387-c20da44a-472a-4905-9074-fed4a71f36e3.jpg)
[2021-07-28 13:03:10.144] [IPC-RENDERER-error] Error: ENOENT: no such file or directory, open '/Users/<user>/Library/Application Support/Defi/.defi/defi.pid'
    at Object.openSync (fs.js:476:3)
    at Object.func [as openSync] (electron/js2c/asar_bundle.js:5:1846)
    at Object.readFileSync (fs.js:377:35)
    at Object.e.readFileSync (electron/js2c/asar_bundle.js:5:8872)
    at Object.getFileData (/Applications/Wallets/DeFi Wallet.app/Contents/Resources/app.asar/electron-app/build/electron-app/src/utils.js:93:23)
    at Function.<anonymous> (/Applications/Wallets/DeFi Wallet.app/Contents/Resources/app.asar/electron-app/build/electron-app/src/services/defiprocessmanager.js:290:39)
    at step (/Applications/Wallets/DeFi Wallet.app/Contents/Resources/app.asar/electron-app/build/electron-app/src/services/defiprocessmanager.js:52:23)
    at Object.next (/Applications/Wallets/DeFi Wallet.app/Contents/Resources/app.asar/electron-app/build/electron-app/src/services/defiprocessmanager.js:33:53)
    at /Applications/Wallets/DeFi Wallet.app/Contents/Resources/app.asar/electron-app/build/electron-app/src/services/defiprocessmanager.js:27:71
    at new Promise (<anonymous>)

**Comment by izzycsy** [comment]
Hi @baludwi , what version were you on previously or is this your first time downloading the DeFi Wallet app?

**Comment by marqueurs404** [comment]
Encountered the same screen on macOS Big Sur version 11.2.2, DeFi Wallet v2.5.2. Did a fresh reinstall following https://github.com/DeFiCh/app/discussions/964 and still had the same issue.

Got the following on debug.log:
```
2021-07-29T15:16:26Z Ignoring unknown configuration value rpcconnect
2021-07-29T15:16:26Z DeFi Blockchain version v1.7.11.0-release (release build)
2021-07-29T15:16:26Z Assuming ancestors of block 9b257cb88630e422902ef2b17a3627ae2f786a5923df9c3bda4226f9551b1ea8 have valid signatures.
2021-07-29T15:16:26Z Setting nMinimumChainWork=000000000000000000000000000000000000000000003f2949bfe4efc275390c
2021-07-29T15:16:26Z Using the 'sse4(1way),sse41(4way)' SHA256 implementation
2021-07-29T15:16:26Z Default data directory /Users/marcusong/Library/Application Support/DeFi
2021-07-29T15:16:26Z Using data directory /Users/marcusong/Library/Application Support/DeFi
2021-07-29T15:16:26Z Config file: /Users/marcusong/Library/Application Support/Defi/.defi/defi.conf
2021-07-29T15:16:26Z Using at most 125 automatic connections (8192 file descriptors available)
2021-07-29T15:16:26Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2021-07-29T15:16:26Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2021-07-29T15:16:26Z Using 8 threads for script verification
2021-07-29T15:16:26Z scheduler thread start
2021-07-29T15:16:26Z Binding RPC on address 127.0.0.1 port 8555 failed.
2021-07-29T15:16:26Z Unable to bind any endpoint for RPC server
2021-07-29T15:16:26Z Error: Unable to start HTTP server. See debug log for details.
2021-07-29T15:16:26Z Shutdown: In progress...
2021-07-29T15:16:26Z scheduler thread interrupt
2021-07-29T15:16:26Z Shutdown: done
```

**Comment by baludwi** [comment]
Same version, 2.5.2. I deinstalled defi-wallet, because it didn't work. But after reinstall the wallet, I got the described error

**Comment by izzycsy** [comment]
Hi @baludwi , I meant before v2.5.2; what version were you on? To uninstall the DeFi Wallet app, you need to remove 3 things: 

1. defi-wallet folder
 
<img width="630" alt="defi-wallet folder location" src="https://user-images.githubusercontent.com/65014479/127739969-ecb82d91-2b7f-4f9d-abc1-7ec0949a5623.png">

2. defi folder

<img width="801" alt="Defi folder" src="https://user-images.githubusercontent.com/65014479/127739974-73f6dd63-c9f2-4623-8c42-56c4688cb111.png">

3. DeFi Wallet app icon

4. There's also a hidden `.defi` folder which you'll need to remove, it should be in the `Defi` folder unless you're one of the early users.

`/Users/username/Library/Application Support/Defi/.defi/` 

Let me know if you can find all 4 items. Remember to backup your `wallet.dat`.

**Comment by baludwi** [comment]
Hi @izzycsy,

before 2.5.2, I came from 2.3.x. I am not sure. I use the defi-wallet since early days in December las year. 

I did all the steps You've described above. Unfortunately it does not work. After reinstall defi-wallet, it takes approx 1 minute while starting followed by the message "Unable to start Blockchain". 

When i cheek the App.log file, I can see this:

`
[2021-08-01 13:19:41.654] [IPC-MAIN-info]  [Starting Electron App] OS macOS Big Sur - 20.5.0
[2021-08-01 13:19:41.662] [IPC-MAIN-info]  Checking for update
[2021-08-01 13:19:41.664] [IPC-MAIN-info]  Wallets folder missing... Creating directory...
[2021-08-01 13:19:42.406] [IPC-MAIN-info]  Generated new staging user ID: 03c91bc4-0f0f-56e4-903e-f3fc851d83d3
[2021-08-01 13:19:42.709] [IPC-MAIN-info]  Update for version 2.5.2 is not available (latest version: 2.5.2, downgrade is disallowed).
[2021-08-01 13:19:42.710] [IPC-MAIN-debug] checkForUpdatesAndNotify called, downloadPromise is null
[2021-08-01 13:19:43.839] [IPC-RENDERER-error]  "Error in selected unit, setting it to default one"
[2021-08-01 13:19:44.024] [IPC-RENDERER-info]   "[Starting App] App is running with version 2.5.2"
[2021-08-01 13:19:44.031] [IPC-RENDERER-info]  [start] - Starting node connection...
[2021-08-01 13:19:44.032] [IPC-RENDERER-info]  Current Node: 1.7.11 New Node: 1.7.11
[2021-08-01 13:19:44.049] [IPC-RENDERER-info]  [start] - Node start initiated
[2021-08-01 13:19:44.052] [IPC-RENDERER-info]  [setLockTimeout] "Starting update of lock timeout..."
[2021-08-01 13:19:44.057] [IPC-RENDERER-info]  [Update Auto-lock Time] "Timeout updated to 300 seconds"
[2021-08-01 13:19:44.189] [IPC-RENDERER-info]  [start] - Node started
[2021-08-01 13:19:44.189] [IPC-RENDERER-info]  [start] - Sending node started to client
[2021-08-01 13:19:44.209] [IPC-RENDERER-error] [start] - Error on node connection: Error: Unable to start HTTP server. See debug log for details.
[2021-08-01 13:19:44.215] [IPC-RENDERER-error] [startBinary] "Error: Unable to start HTTP server. See debug log for details."
[2021-08-01 13:19:44.218] [IPC-RENDERER-info]  [Blockchain Status] {"success":false,"data":{"message":"Error: Unable to start HTTP server. See debug log for details.","walletMap":{"paths":["/Users/<UserName>/Library/Application Support/Defi/wallets/wallet.dat"],"nodeVersion":"1.7.11","hasSyncSPV":false,"hasRescan":false}}}
[2021-08-01 13:19:44.220] [IPC-RENDERER-info]  [start] - Node is closed with code 1
[2021-08-01 13:19:44.221] [IPC-RENDERER-error] [start] - Error occurred while closing node with code: 1
[2021-08-01 13:19:44.222] [IPC-RENDERER-error] [blockChainNotStarted - Not Running] ""
[2021-08-01 13:19:44.232] [IPC-RENDERER-error] [startBinary] "Error occurred while closing node with code: 1"
[2021-08-01 13:19:44.234] [IPC-RENDERER-info]  [Blockchain Status] {"success":false,"data":{"message":"Error occurred while closing node with code: 1","walletMap":{"paths":["/Users/<UserName>/Library/Application Support/Defi/wallets/wallet.dat"],"nodeVersion":"1.7.11","hasSyncSPV":false,"hasRescan":false}}}
[2021-08-01 13:19:44.236] [IPC-RENDERER-error] [blockChainNotStarted - Not Running] ""
[2021-08-01 13:19:46.222] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:48.206] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:50.205] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:52.199] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:54.205] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:56.205] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:19:58.205] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:00.205] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:02.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:04.199] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:06.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:08.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:10.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:12.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:14.204] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:16.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:18.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:20.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:22.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:24.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:26.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:28.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:30.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:32.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:34.303] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:36.309] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:38.306] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:40.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:42.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:44.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:46.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:48.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:50.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:52.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:54.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:56.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:20:58.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:00.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:02.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:04.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:06.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:08.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:10.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:12.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:14.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:16.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:18.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:20.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:22.306] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:24.307] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:26.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:28.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:30.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:32.306] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:34.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:36.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:38.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:40.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:42.305] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:44.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:46.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:48.308] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:50.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:52.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:54.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:56.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:21:58.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:00.311] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:02.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:04.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:06.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:08.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:10.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:12.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:14.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:16.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:18.310] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:20.312] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:22.308] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:24.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:26.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:28.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:30.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:32.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:34.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:36.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:38.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:40.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:42.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:44.314] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:46.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:48.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:50.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:52.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:54.313] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:56.314] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:22:58.314] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:23:00.314] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:23:02.308] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:23:04.314] [IPC-RENDERER-error] [isBlockchainStarted] "Request failed with status code 503"
[2021-08-01 13:23:04.316] [IPC-RENDERER-info]  [Blockchain Status] "Unable to start blockchain"
[2021-08-01 13:23:04.318] [IPC-RENDERER-error] [blockChainNotStarted - Not Running] "Unable to start blockchain"
`

**Comment by baludwi** [comment]
After restart MacOS, this issue is obsolete. Nomally, a Mac doesn't has to restarted, but ok. Now defi-wallet can be started normal.
